sources/constraint-declaration-validation.asciidoc

```diff
@@ -1026,6 +1100,289 @@ public class SimpleOrderService extends OrderService {
 
 The return value constraints in [classname]`DefaultOrderService` are legal, as they strengthen the postconditions of the [methodname]`placeOrder()` method as constituted by the super class [classname]`OrderService` but don't weaken them.
 
+[[constraintdeclarationvalidationprocess-containerelementconstraints]]
+=== Container element constraints
+
+Constraints can be applied to the elements of generic containers, e.g. `List`, `Map` or `Optional`.
+This is done by putting constraint annotations to the type arguments of such containers.
+Container element constraints can be applied to containers that are
+
+* fields,
+* properties,
+* method or constructor parameters or
+* method return values.
+
+.Container element constraints
+====
+[source, JAVA]
+----
+private List<@Email String> emails;
+
+public Optional<@Email String> getEmail() {
+    [...]
+}
+
+public Map<@NotNull String, @ValidAddress Address> getAddressesByType() {
+    [...]
+}
+
+public List<@NotBlank String> getMatchingRecords(List<@NotNull @Size(max=20) String> searchTerms) {
+    [...]
+}
+----
+====
+
+When a field, property, executable parameter or method return value is validated,
+any container element constraints of that element will be validated alongside any other constraints hosted by that element.
+For container element constraints, the same rules for validation groups and group sequences apply as for any other constraint on the same element.
+
+When a container element constraint is detected, the validation engine invokes a value extractor (see <<constraintdeclarationvalidationprocess-valueextractordefinitions>>)
```

Comment:
s/container element constraint is detected/container element constraint is constrained/

Because you don't want to described a process in a spec if you can express a behavior. For example from your sentence I could infer that I need to extract for each constraint (i.e. 2 constraints on a container element means double extraction).